packages/vega-scenegraph/src/CanvasRenderer.js

```diff
@@ -31,9 +36,16 @@ prototype.initialize = function(el, width, height, origin, scaleFactor, options)
 };
 
 prototype.resize = function(width, height, origin, scaleFactor) {
+  if (this._resized) {
+    throw new Error('cannot resize canvas twice');
+  }
+
+  this._resized = true;
```

Comment:
I'd prefer to have the logic prevent this from happening, alongside any relevant test cases, rather than add a new member variable and check an assertion at runtime.

Reply:
OK